backend/store/db_connection.go

```diff
@@ -79,10 +83,25 @@ func (m *DBConnectionManager) Close() error {
 		m.watcher.Close()
 	}
 
-	if m.db != nil {
-		return m.db.Close()
+	m.mu.Lock()
```

Comment:
Can we just add a single line `stdlib.UnregisterConnConfig(m.connStr)` in line 81 in the `Close`? 
Or
```golang
m.mu.Lock()
defer func() {
  m.mu.Unlock()
}

stdlib.UnregisterConnConfig(m.connStr)
if m.db != nil {
  return m.db.Close()
}
```